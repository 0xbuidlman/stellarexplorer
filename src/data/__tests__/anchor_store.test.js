import AnchorStore from '../anchor_store'

jest.mock('fetch-ponyfill')

const feedUrl = testFile => `https://feed/anchor_store.test.${testFile}.json`

it('pulls anchors list', done => {
  const store = new AnchorStore(feedUrl('1'), 'assets')

  setTimeout(() => {
    const anchors = store.getAnchors()
    expect(Object.keys(anchors).length).toEqual(1)
    expect(anchors['vcbear.net']).toMatchSnapshot()
    done()
  }, 150)
})
