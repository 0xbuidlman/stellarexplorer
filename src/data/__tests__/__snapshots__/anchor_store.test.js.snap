// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pulls anchors list 1`] = `
Array [
  Object {
    "activityScore": 10.917,
    "code": "JPY",
    "depth10_USD": 5027.37,
    "depth10_XLM": 220045,
    "domain": "vcbear.net",
    "id": "JPY-GBVAOIACNSB7OVUXJYC5UE2D4YK2F7A24T7EE5YOMN4CE6GCHUTOUQXM",
    "issuer": "GBVAOIACNSB7OVUXJYC5UE2D4YK2F7A24T7EE5YOMN4CE6GCHUTOUQXM",
    "numAsks": 20,
    "numBids": 20,
    "numTrades24h": 111,
    "price_USD": 0.0098641,
    "price_XLM": 0.43175,
    "slug": "JPY-vcbear.net",
    "spread": 0.009,
    "topTradePairSlug": "XLM-native/JPY-vcbear.net",
    "volume24h_USD": 15507.47,
    "volume24h_XLM": 678753.12,
    "website": "https://vcbear.net/",
  },
]
`;
